semgrep/rxjs-syntax.yaml

```diff
@@ -2,11 +2,30 @@ rules:
 
   - id: combineLatest-not-combining
     pattern: combineLatest([$SINGLE_EXPRESSION])
-    fix:  $SINGLE_EXPRESSION.
-    message: replace combineLatest with a single operand is not useful
-    metadata: {
-      todo: "remove dot in fix once bug is fixed (https://github.com/returntocorp/semgrep/issues/1721)"
-      }
+    fix:  $SINGLE_EXPRESSION
+    message: combineLatest is not needed with a single argument
     languages: [ts]
     severity: WARNING
- 
\ No newline at end of file
+
+  - id: observables-from-select-or-pipe-not-ending-with-dollar-sign
+    patterns:
+    - pattern-regex: '\w+\s*(:.+?)?=.*?(select|pipe)'
+    - pattern-either:
+      - pattern-inside: $VAR = $EXPR.select(...);
+      - pattern-inside: $VAR = $EXPR.pipe(...);
```

Comment:
Semgrep only resolves down to the level of language constructs. The characters that make up a variable name, though, are below that. The regex, therefore, is used to find identifiers consisting exclusively of word characters (possibly followed by a type designation)... and thus _without_ a trailing dollar sign.